_overviews/toolkit/web-server-dynamic.md

```diff
@@ -0,0 +1,228 @@
+  @cask.get("/time")
+  def dynamic() = s"Current date is: ${ZonedDateTime.now()}"
+
+  initialize()
+```
+{% endtab %}
+{% endtabs %}
+
+
+Run the example the same way as before (assuming you use the same project structure).
+
+{% tabs web-server-dynamic-2 class=tabs-build-tool %}
+{% tab 'Scala CLI' %}
+In the terminal, the following command will start the server:
+```
+scala-cli run Example.sc
+```
+{% endtab %}
+{% tab 'sbt' %}
+In the sbt shell, the following command will start the server:
+```
+sbt:example> example/run
+```
+{% endtab %}
+{% tab 'Mill' %}
+In the terminal, the following command will start the server:
+```
+./mill run
+```
+{% endtab %}
+{% endtabs %}
+
+Access the endpoint at [http://localhost:8080/time](http://localhost:8080/time). You should see a result similar to the
+one below.
+
+```
+Current date is: 2024-07-22T09:07:05.752534+02:00[Europe/Warsaw]
+```
+
+## Using path segments
+
+You can use path segments to specify the returned data more precisely. Building on the example above, add the `:city`
+segment to get the current time in a city of choice.
+
+{% tabs web-server-dynamic-3 class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+import java.time.{ZoneId, ZonedDateTime}
+
+object MyApp extends cask.MainRoutes {
+
+  private def getZoneIdForCity(city: String): Option[ZoneId] = {
+    import scala.jdk.CollectionConverters._
+    ZoneId.getAvailableZoneIds.asScala.find(_.endsWith("/" + city)).map(ZoneId.of)
+  }
+
+  @cask.get("/time/:city")
+  def dynamicWithCity(city: String) = {
+    getZoneIdForCity(city) match {
+      case Some(zoneId) => s"Current date is: ${ZonedDateTime.now().withZoneSameInstant(zoneId)}"
+      case None => s"Couldn't find time zone for city $city"
+    }
+  }
+
+  initialize()
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
+```scala
+import java.time.{ZoneId, ZonedDateTime}
+
+object MyApp extends cask.MainRoutes:
+
+  private def getZoneIdForCity(city: String): Option[ZoneId] =
+    import scala.jdk.CollectionConverters.*
+    ZoneId.getAvailableZoneIds.asScala.find(_.endsWith("/" + city)).map(ZoneId.of)
+  
+  @cask.get("/time/:city")
+  def dynamicWithCity(city: String) =
+    getZoneIdForCity(city) match
+      case Some(zoneId) => s"Current date is: ${ZonedDateTime.now().withZoneSameInstant(zoneId)}"
+      case None => s"Couldn't find time zone for city $city"
+
+  initialize()
+```
+{% endtab %}
+{% endtabs %}
+
+Accessing the endpoint at [http://localhost:8080/time/Paris](http://localhost:8080/time/Paris) will result with:
+```
+Current date is: 2024-07-22T09:08:33.806259+02:00[Europe/Paris]
+```
+
+and at [http://localhost:8080/time/Tokyo](http://localhost:8080/time/Tokyo) you will see:
+```
+Current date is: 2024-07-22T16:08:41.137563+09:00[Asia/Tokyo]
+```
+
+Cask endpoints can handle either fixed or arbitrary number of path segments. Please consult the 
+[documentation](https://com-lihaoyi.github.io/cask/index.html#variable-routes) for more details.
```

Comment:
The first sentence here is a bit obscure. Consider leaving only the second sentence.

Also, "Please consult" -> "Consult".

Reply:
I rewrote it to something hopefully more understandable.